react-native/react/login2/register/set-public-name/index.js

```diff
@@ -15,11 +15,14 @@ export default class SetPublicName extends Component {
     }
   }
 
-  submit () {
-    this.props.submit(this.state.deviceName)
+  onSubmit () {
+    this.props.onSubmit(this.state.deviceName)
   }
 
   render () {
+    const dupeName = this.props.existingDevices && this.props.existingDevices.indexOf(this.state.deviceName) !== -1
```

Comment:
Also, is there any sign that you've typed a dupe name other than the button being disabled?